core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -9,55 +9,503 @@
+ * ---------------------------------------------------------------------------------------
+ * 6 | Key Reference          | Reference encoding all info needed to           |
+ *   |                        | access the key off-heap                         |
+ * -----------------------------------------------------------------------------| entry with
+ * 7 | Value Reference        | Reference encoding all info needed to           | entry index
+ *   |                        | access the key off-heap.                        | 2
+ * -----------------------------------------------------------------------------|
+ * 8 | Optional additional entry's 2 field, set by derived class                |
+ * ---------------------------------------------------------------------------------------
+ * ...
+ *
+* */
 abstract class BasicChunk<K, V> {
+    /**
+     * offset in primitive fields of one entry, to a reference as coded by
+     * keysMemoryManager, of the key pointed from this entry (size of long)
+     */
+    protected static final int KEY_REF_OFFSET = 0;
+
+    /**
+     * offset in primitive fields of one entry, to a reference as coded by
+     * valuesMemoryManager, of the value pointed from this entry (size of long).
+     * The primitive field equals to INVALID_REFERENCE if no value is point.
+     */
+    protected static final int VALUE_REF_OFFSET = 1;
+
+    protected static final int INVALID_ENTRY_INDEX = -1;
+
+    protected final OakSharedConfig<K, V> config;
+
+    protected final MemoryManager valuesMemoryManager;
+    protected final MemoryManager keysMemoryManager;
+
+    protected final EntryArray array;
+
+    // Counts number of entries inserted & not deleted. Pay attention that not all entries (counted
+    // in number of entries) are finally considered existing by the OrderedChunk above
+    // and participating in holding the "real" KV-mappings, the "real" are counted in OrderedChunk
+    protected final AtomicInteger numOfEntries;
+
+    // for writing the keys into the off-heap
+    final OakSerializer<K> keySerializer;
+    final OakSerializer<V> valueSerializer;
+    // to compare serilized and object keys
+    protected final OakComparator<K> comparator;
 
-    /*-------------- Constants --------------*/
+
+    /*-------------- Chunks fields --------------*/
     enum State {
         INFANT,
         NORMAL,
         FROZEN,
         RELEASED
     }
 
-    /*-------------- Members --------------*/
-    // to compare serilized and object keys
-    protected OakComparator<K> comparator;
     // in split/compact process, represents parent of split (can be null!)
     private final AtomicReference<BasicChunk<K, V>> creator;
     // chunk can be in the following states: normal, frozen or infant(has a creator)
     private final AtomicReference<State> state;
     private final AtomicReference<Rebalancer<K, V>> rebalancer;
     private final AtomicInteger pendingOps;
-    private final int maxItems;
```

Comment:
Why maxItems is removed?

Reply:
Because it can be derived from the entry array capacity.

Reply:
Then it is better to keep `getMaxItems()` method and redirect to array capacity